apps/prairielearn/src/pages/clientFilesQuestion/clientFilesQuestion.js

```diff
@@ -1,28 +1,32 @@
+// @ts-check
+const asyncHandler = require('express-async-handler');
 const path = require('path');
 const express = require('express');
 const router = express.Router();
 
 const error = require('@prairielearn/error');
 const chunks = require('../../lib/chunks');
-const ERR = require('async-stacktrace');
+const { getQuestionCourse } = require('../../lib/question');
 
-router.get('/*', function (req, res, next) {
-  const filename = req.params[0];
-  if (!filename) {
-    return next(
-      error.make(400, 'No filename provided within clientFilesQuestion directory', {
+router.get(
+  '/*',
+  asyncHandler(async function (req, res) {
+    const filename = req.params[0];
+    if (!filename) {
+      throw error.make(400, 'No filename provided within clientFilesQuestion directory', {
         locals: res.locals,
         body: req.body,
-      }),
-    );
-  }
-  const coursePath = chunks.getRuntimeDirectoryForCourse(res.locals.course);
-  const chunk = {
-    type: 'question',
-    questionId: res.locals.question.id,
-  };
-  chunks.ensureChunksForCourse(res.locals.course.id, chunk, (err) => {
-    if (ERR(err, next)) return;
+      });
+    }
+    const course = await getQuestionCourse(res.locals.question, res.locals.course);
```

Comment:
Refresh my memory: by the time we get to this point, we've correctly authenticated the the current user/course has access to this question via sharing, correct? I think that's true, but I always like to triple-check with auth code.

Reply:
That's right, auth is all taken care of. 

Whenever one of these URLs is accessed through an `instance_question`, which is any time a student accesses it, the instance question was created from an assessment question, which was created at sync time after validating that the course has permissions to use this question.

If an instructor accesses this through a link that is not associated with a question instance (say, in a question preview), then we validate in the middleware that the question has been shared with the course.